pkg/kudoctl/util/repo/repo_operator.go

```diff
@@ -79,21 +79,43 @@ func (r *Client) DownloadIndexFile() (*IndexFile, error) {
 }
 
 // getPackageReaderByFullPackageName downloads the tgz file from the remote repository and unmarshals it to the package CRDs
-func (r *Client) getPackageReaderByFullPackageName(fullPackageName string) (io.Reader, error) {
-	var fileURL string
-	parsedURL, err := url.Parse(r.Config.URL)
+// this will cycle through urls for the
+func (r *Client) getPackageReaderByFullPackageName(pkg *PackageVersion) (io.Reader, error) {
```

Comment:
this is no longer getting anything by full package name, right? Or at least it's not that obvious. Could we maybe go with `getPackageReader` only?

Reply:
also ... who is responsible for closing the reader? It looks like it just flows down the code and nobody really owns it and closes it...

Reply:
that is a fantastic point... an issue that has been here for some time!